synapse/util/caches/deferred_cache.py

```diff
@@ -104,7 +104,13 @@ def metrics_cb() -> None:
             max_size=max_entries,
             cache_name=name,
             cache_type=cache_type,
-            size_callback=(lambda d: len(d) or 1) if iterable else None,
+            size_callback=(
+                (lambda d: len(d) or 1)  # type: ignore[arg-type]
```

Comment:
wonder how this compares on a 'unfortunate but hard to work around':

```suggestion
                (lambda d: len(cast(Sized, d)) or 1)
```

I *think* (but don't feel too strongly) that I prefer the cast here, mostly because it shows exactly what you need to override, whereas the comment makes me have to think about which 'arg' it's talking about.